src/test/regress/expected/failure_single_select.out

```diff
@@ -81,6 +81,7 @@ DETAIL:  server closed the connection unexpectedly
 (3 rows)
 
 COMMIT;
+ERROR:  failure on connection marked as essential: localhost:xxxxx
```

Comment:
So this is an odd case, maybe we'd like to let this COMMIT go through. What's happening is that the SELECT occurs on the same connection as the INSERT, we create a new connection for the second INSERT, but when we go to commit we detect that the first INSERT's connection failed. But it failed while doing something non essential